src/main/java/com/facebook/openwifirrm/ucentral/models/AggregatedState.java

```diff
@@ -0,0 +1,214 @@
+				return;
+			}
+			if (mcs == null || mcs.isEmpty()) {
+				this.bitrate = rate.bitrate;
+				this.chwidth = rate.chwidth;
+			}
+			this.mcs.addAll(rate.mcs);
+		}
+	}
+
+	public String bssid;
+	public String station;
+	public long connected;
+	public long inactive;
+	public List<Integer> rssi;
+	public long rx_bytes;
+	public long rx_packets;
+	public AggregatedRate rx_rate;
+	public long tx_bytes;
+	public long tx_duration;
+	public long tx_failed;
+	public long tx_packets;
+	public AggregatedRate tx_rate;
+	public long tx_retries;
+	public int ack_signal;
+	public int ack_signal_avg;
+	public Radio radio;
+
+	public class Radio {
+		public int channel;
+		public int channel_width;
+		public int tx_power;
+
+		public Radio() {}
+
+		public Radio(int channel, int channel_width, int tx_power) {
+			this.channel = channel;
+			this.channel_width = channel_width;
+			this.tx_power = tx_power;
+		}
+
+		public Radio(int[] radios) {
+			this.channel = radios[0];
+			this.channel_width = radios[1];
+			this.tx_power = radios[2];
+		}
+
+		@Override
+		public int hashCode() {
+			return Objects.hash(channel, channel_width, tx_power);
+		}
+
+		@Override
+		public boolean equals(Object obj) {
+			if (this == obj) {
+				return true;
+			}
+			if (obj == null) {
+				return false;
+			}
+			if (getClass() != obj.getClass()) {
+				return false;
+			}
+
+			Radio other = (Radio) obj;
+			return channel == other.channel &&
+				channel_width == other.channel_width &&
+				tx_power == other.tx_power;
+		}
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(bssid, station, radio);
+	}
+
+	/** Constructor with no args */
+	public AggregatedState() {
+		this.rx_rate = new AggregatedRate();
+		this.tx_rate = new AggregatedRate();
+		this.rssi = new ArrayList<>();
+		this.radio = new Radio();
+	}
+
+	/** Construct from Aggregatedstate */
+	public AggregatedState(AggregatedState state) {
+		this.rx_rate = new AggregatedRate();
+		this.tx_rate = new AggregatedRate();
+		this.rssi = new ArrayList<>();
+		add(state);
+	}
+
+	/** Construt from Asscociation and radio */
+	public AggregatedState(Association association, int[] radios) {
+		this.rx_rate = new AggregatedRate();
+		this.tx_rate = new AggregatedRate();
+		this.rssi = new ArrayList<>();
+
+		this.bssid = association.bssid;
+		this.station = association.station;
+		this.connected = association.connected;
+		this.inactive = association.inactive;
+		this.rssi.add(association.rssi);
+		this.rx_bytes = association.rx_bytes;
+		this.rx_packets = association.rx_packets;
+		this.rx_rate.add(association.rx_rate);
+		this.tx_bytes = association.tx_bytes;
+		this.tx_duration = association.tx_duration;
+		this.tx_failed = association.tx_failed;
+		this.tx_packets = association.tx_packets;
+		this.tx_rate.add(association.tx_rate);
+		this.tx_retries = association.tx_retries;
+		this.ack_signal = association.ack_signal;
+		this.ack_signal_avg = association.ack_signal_avg;
```

Comment:
camel case? the original structure is only snake case so that the json gets parsed appropriately